Sources/SimultaneouslyScrollView/Extensions/MulticastScrollViewDelegate.swift

```diff
@@ -0,0 +1,107 @@
+import UIKit;
+
+class MulticastScrollViewDelegate: NSObject, UIScrollViewDelegate {
+    // Store weak references to delegates
+    private var delegates = NSHashTable<UIScrollViewDelegate>.weakObjects()
+
+    func addDelegate(_ delegate: UIScrollViewDelegate) {
+        delegates.add(delegate)
+    }
+
+    func removeDelegate(_ delegate: UIScrollViewDelegate) {
+        delegates.remove(delegate)
+    }
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidScroll?(scrollView)
+        }
+    }
+    
+
+    func scrollViewDidZoom(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidZoom?(scrollView)
+        }
+    }
+
+    func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewWillBeginDragging?(scrollView)
+        }
+    }
+
+    func scrollViewWillEndDragging(_ scrollView: UIScrollView, withVelocity velocity: CGPoint, targetContentOffset: UnsafeMutablePointer<CGPoint>) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewWillEndDragging?(scrollView, withVelocity: velocity, targetContentOffset: targetContentOffset)
+        }
+    }
+
+    func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidEndDragging?(scrollView, willDecelerate: decelerate)
+        }
+    }
+
+    func scrollViewWillBeginDecelerating(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewWillBeginDecelerating?(scrollView)
+        }
+    }
+
+    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidEndDecelerating?(scrollView)
+        }
+    }
+
+    func scrollViewDidEndScrollingAnimation(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidEndScrollingAnimation?(scrollView)
+        }
+    }
+
+    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+        for delegate in delegates.allObjects {
+            if let view = delegate.viewForZooming?(in: scrollView) {
+                return view
+            }
+        }
+        return nil
```

Comment:
Do you think this might eventually become a problem? What if multiple delegates define a different return value? 🤔 Basically holds for all delegate methods with a return value.

Reply:
Well, it's definitely not a problem now, as we don't expose the multicast delegate class, and we know for a fact that your code doesn't use that delegate method, so there's at most one delegate implementing it.